src/ofxGLWarper.cpp

```diff
@@ -213,76 +194,74 @@ void ofxGLWarper::loadFromXml(ofXml &XML, const string& warperID){
-        float distx = corners[i].x - args.x;
-        float disty = corners[i].y - args.y;
-        float dist  = sqrt( distx * distx + disty * disty);
+        float distx = corners[i]->x - args.x;
+        float disty = corners[i]->y - args.y;
+        float dist  = sqrtf( distx * distx + disty * disty);
         ofLogVerbose() << "mouse to corner dist: " << dist << endl;
         if(dist < smallestDist && dist < sensFactor ){
-            whichCorner = i;
+            selectedCorner = i;
             smallestDist = dist;
-            cornerSelected=true;
+            cornerIsSelected=true;
         }
     }
 }
 
 //--------------------------------------------------------------
 void ofxGLWarper::keyPressed(ofKeyEventArgs &args){
-	if (whichCorner >= 0 && cornerSelected) {
+    if (cornerIsSelected && selectedCorner >= 0) {
         switch (args.key) {
             case OF_KEY_DOWN:
-                corners[whichCorner].y++;
+                corners[selectedCorner] += glm::vec2(0,1);
+                processMatrices();
                 break;
             case OF_KEY_UP:
-                corners[whichCorner].y--;
+                corners[selectedCorner] += glm::vec2(0,-1);
+                processMatrices();
                 break;
             case OF_KEY_LEFT:
-                corners[whichCorner].x--;
+                corners[selectedCorner] += glm::vec2(-1,0);
+                processMatrices();
                 break;
             case OF_KEY_RIGHT:
-                corners[whichCorner].x++;
+                corners[selectedCorner] += glm::vec2(1,0);
+                processMatrices();
                 break;
             default:
 			break;
 		}
 	}
 }
-//--------------------------------------------------------------
-ofVec4f ofxGLWarper::fromScreenToWarpCoord(float x, float y, float z){
-    ofVec4f mousePoint;
-    ofVec4f warpedPoint;
+
+ //--------------------------------------------------------------
+glm::vec4 ofxGLWarper::fromScreenToWarpCoord(float x, float y, float z){
+    glm::vec4 mousePoint;
+    glm::vec4 warpedPoint;
 
     // this is the point on the image which i want to know the coordinates inside the warped system ...
     mousePoint.x = x;
     mousePoint.y = y;
-    mousePoint.z = 0.0;
+    mousePoint.z = z;
     mousePoint.w = 1.0;
 
-    // i create a ofMatrix4x4 with the ofxGLWarper myMatrixData in column order
-    ofMatrix4x4 myOFmatrix = ofMatrix4x4::getTransposedOf(myMatrix);
-
-    // do not invert the matrix
-    ofMatrix4x4 invertedMyMatrix = myOFmatrix.getInverse();
-    //ofMatrix4x4 invertedMyMatrix = myOFmatrix;
+    // with glm matrices, no need to transpose and inverse, because if you have a mat4 and do mat4*vec4,
+    // your vec4 is considered a column vector. If you do vec4*mat4, it is considered a row vector.
 
     // multiply both to get the point transformed by the matrix
-    warpedPoint = invertedMyMatrix * mousePoint ;
+    warpedPoint = myMatrix * mousePoint ;
```

Comment:
of Course !!! It was not very easy, but I got all working like before, I wouldn't say improved, but up to date. Also less lines needed with glm. As I commented:
    // with glm matrices, no need to transpose and inverse, because if you have a mat4 and do mat4*vec4,
    // your vec4 is considered a column vector. If you do vec4*mat4, it is considered a row vector.